src/tests/components/LoginContainer.test.js

```diff
@@ -0,0 +1,203 @@
+  //   const props = {
+  //     onChange: jest.fn(),
+  //     emptyEmail: jest.fn(),
+  //     handleSubmit: jest.fn(),
+  //     checkValidEmail: jest.fn(),
+  //     checkPassword: jest.fn(),
+  //     loginAction: jest.fn(),
+  //   };
+  //   // const wrapper = shallow(
+  //   //   <LoginContainer {...props} />,
+  //   // );
+  //   wrapper.setState({
+  //     email: 'fkiryowa@gmail.com',
+  //   });
+  //   const instance = wrapper.instance();
+  //   const checkValidEmail = jest.spyOn(instance, 'checkValidEmail');
+  //   instance.handleSubmit({ preventDefault: jest.fn() });
+  //   expect(checkValidEmail).toHaveBeenCalled();
+  // });
+
+  // it('should call a submit a form', () => {
+  //   const props = {
+  //     onChange: jest.fn(),
+  //     emptyEmail: jest.fn(),
+  //     handleSubmit: jest.fn(),
+  //     checkValidEmail: jest.fn(),
+  //     checkPassword: jest.fn(),
+  //     loginAction: jest.fn(),
+  //   };
+  //   // const wrapper = shallow(
+  //   //   <LoginContainer {...props} />,
+  //   // );
+  //   wrapper.setState({
+  //     email: 'fkiryowa',
+  //     errors: { userEmail: 'Password field is empty' }
+  //   });
+  //   const instance = wrapper.instance();
+  //   const checkValidEmail = jest.spyOn(instance, 'checkValidEmail');
+  //   instance.handleSubmit({ preventDefault: jest.fn() });
+  //   expect(checkValidEmail).toHaveBeenCalled();
+  // });
+
+  // it('undefined email', () => {
+  //   const props = {
+  //     onChange: jest.fn(),
+  //     emptyEmail: jest.fn(),
+  //     handleSubmit: jest.fn(),
+  //     checkValidEmail: jest.fn(),
+  //     checkPassword: jest.fn(),
+  //     loginAction: jest.fn(),
+  //   };
+  //   // const wrapper = shallow(
+  //   //   <LoginContainer {...props} />,
+  //   // );
+  //   wrapper.setState({
+  //     email: undefined,
+  //     errors: { userEmail: 'Email required' },
+  //   });
+  //   const instance = wrapper.instance();
+  //   instance.handleSubmit({ preventDefault: jest.fn() });
+  //   expect('Email required').toEqual(instance.state.errors.userEmail);
+  // });
+
+  // it('Test everything is fine', () => {
+  //   const props = {
+  //     onChange: jest.fn(),
+  //     emptyEmail: jest.fn(),
+  //     handleSubmit: jest.fn(),
+  //     checkValidEmail: jest.fn(),
+  //     checkPassword: jest.fn(),
+  //     loginAction: jest.fn(),
+  //   };
+  //   // const wrapper = shallow(
+  //   //   <LoginContainer {...props} />,
+  //   // );
+  //   wrapper.setState({
+  //     email: 'fkiryowa@gmail.com',
+  //     password: 'Test@12345',
+  //     errors: { },
+  //   });
+  //   const instance = wrapper.instance();
+  //   instance.handleSubmit({ preventDefault: jest.fn() });
+  //   expect(props.loginAction).toHaveBeenCalled();
+  // });
+
+  // it('Testing mapStateToProps', () => {
+  //   const initialState = {
+  //     auth: {},
+  //   };
+  //   expect(mapStateToProps(initialState)).toEqual({
+  //     login: {},
+  //   });
+  // });
+});
```

Comment:
Even this file has a lot of commented code

Reply:
Done.